_data-prepper/pipelines/configuration/sinks/s3.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: s3
+parent: Sinks
+grand_parent: Pipelines
+nav_order: 55
+---
+
+# s3
+
+The `s3` sink sends records to an Amazon S3 bucket using the S3 client.
+
+## Usage
+
+The following example creates a pipeline configured with an s3 sink. It contains additional options to customize the event and size thresholds for which the pipeline sends record events, and sets the codec type `ndjson`.
+
+```
+pipeline:
+  ...
+  sink:
+    - s3:
+        aws:
+          region: us-east-1
+          sts_role_arn: arn:aws:iam::123456789012:role/Data-Prepper
+          sts_header_overrides:
+        max_retries: 5
+        bucket:
+          name: bucket_name
+          object_key:
+            path_prefix: my-elb/%{yyyy}/%{MM}/%{dd}/
+        threshold:
+          event_count: 2000
+          maximum_size: 50mb
+          event_collect_timeout: 15s
+        codec:
+          ndjson:
+        buffer_type: in_memory
+```
+
```

Comment:
We are missing a top-level configuration section. We should add that, and it should indicate what options are available. I have a few other comments about other items that should get moved up there.

See the S3 source for the current pattern to use.